src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeNamespaceScopeEditor.cs

```diff
@@ -0,0 +1,151 @@
+                TNamespaceDeclarationSyntax namespaceDeclaration,
+                TTypeDeclarationSyntax typeToMove,
+                Document documentToEdit,
+                CancellationToken cancellationToken)
+            {
+                var syntaxFactsService = documentToEdit.GetLanguageService<ISyntaxFactsService>();
+                var childNodes = syntaxFactsService.GetMembersOfNamespaceDeclaration(namespaceDeclaration);
+
+                if (childNodes.Count <= 1)
+                {
+                    return null;
+                }
+
+                var editor = await DocumentEditor.CreateAsync(documentToEdit, cancellationToken).ConfigureAwait(false);
+                editor.RemoveNode(typeToMove, SyntaxRemoveOptions.KeepNoTrivia);
+
+                var syntaxGenerator = editor.Generator;
+                var index = childNodes.IndexOf(typeToMove);
+
+                var itemsBefore = index > 0 ? childNodes.Take(index) : Enumerable.Empty<SyntaxNode>();
+                var itemsAfter = index < childNodes.Count - 1 ? childNodes.Skip(index + 1) : Enumerable.Empty<SyntaxNode>();
+
+                var name = syntaxFactsService.GetDisplayName(namespaceDeclaration, DisplayNameOptions.IncludeNamespaces);
+                var newNamespaceDeclaration = syntaxGenerator.NamespaceDeclaration(name, WithElasticTrivia(typeToMove)).WithAdditionalAnnotations(NamespaceScopeMovedAnnotation);
+
+                if (itemsBefore.Any() && itemsAfter.Any())
+                {
+                    var itemsAfterNamespaceDeclaration = syntaxGenerator.NamespaceDeclaration(name, WithElasticTrivia(itemsAfter));
+
+                    foreach (var nodeToRemove in itemsAfter)
+                    {
+                        editor.RemoveNode(nodeToRemove, SyntaxRemoveOptions.KeepNoTrivia);
+                    }
+
+                    editor.InsertAfter(namespaceDeclaration, new[] { newNamespaceDeclaration, itemsAfterNamespaceDeclaration });
+                }
+                else if (itemsBefore.Any())
+                {
+                    editor.InsertAfter(namespaceDeclaration, newNamespaceDeclaration);
+
+                    var nodeToCleanup = itemsBefore.Last();
+                    editor.ReplaceNode(nodeToCleanup, WithElasticTrivia(nodeToCleanup, leading: false));
+                }
+                else if (itemsAfter.Any())
+                {
+                    editor.InsertBefore(namespaceDeclaration, newNamespaceDeclaration);
+
+                    var nodeToCleanup = itemsAfter.First();
+                    editor.ReplaceNode(nodeToCleanup, WithElasticTrivia(nodeToCleanup, trailing: false));
+                }
+
+                var changedDocument = editor.GetChangedDocument();
+                return changedDocument.Project.Solution;
+            }
+
+            private static SyntaxNode WithElasticTrivia(SyntaxNode syntaxNode, bool leading = true, bool trailing = true)
```

Comment:
seems good one to share? put it with other trivia helper methods ?

Reply:
Sure